fe/fe-core/src/main/java/org/apache/doris/persist/EditLog.java

```diff
@@ -1345,6 +1347,36 @@ private <T extends Writable> void logEdit(short op, List<T> entries) throws IOEx
         }
     }
 
+    // NOTICE: No guarantee atomicity of entries
+    private <T extends Writable> void logEdit(List<Pair<Short, T>> entries) throws IOException {
+        int itemNum = Math.max(1, Math.min(Config.batch_edit_log_max_item_num, entries.size()));
+        JournalBatch batch = new JournalBatch(itemNum);
+        long batchCount = 0;
+        for (Pair<Short, T> entry : entries) {
+            if (batch.getJournalEntities().size() >= Config.batch_edit_log_max_item_num
+                    || batch.getSize() >= Config.batch_edit_log_max_byte_size) {
+                journal.write(batch);
+                batch = new JournalBatch(itemNum);
```

Comment:
Please don't split the batch if you need to get an atomic guarantee.

You can change the signature of this method to

```
void logEdit(JournalBatch batch) throws IOException
```